File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let max_area = 0;
    let start = 0;
    let end = height.length - 1

    while(start < end){
        const current_area = Math.min(height[start], height[end]) * (end - start) // height * width

        if (height[start] <= height[end]){
            start += 1
        } else {
            end -= 1
        }
        max_area = Math.max(max_area, current_area)
    }
    return max_area;
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        max_w = 0

        i = 0
        j = len(height) - 1

        while i <= j:
            area = min(height[i], height[j]) * (j-i)
            if height[i] <= height[j]:
                i += 1
            else:
                j -= 1

            max_w = max(area, max_w)

        return max_w


